interop-testing/src/test/java/io/grpc/testing/integration/StressTestClientTest.java

```diff
@@ -0,0 +1,164 @@
+    StressTestClient client = new StressTestClient();
+    client.parseArgs(new String[] {"--server_addresses=0:0:0:0:0:0:0:1:8080,1:2:3:4:f:e:a:b:8083"});
+
+    assertEquals(2, client.addresses().size());
+    assertEquals(new InetSocketAddress("0:0:0:0:0:0:0:1", 8080), client.addresses().get(0));
+    assertEquals(new InetSocketAddress("1:2:3:4:f:e:a:b", 8083), client.addresses().get(1));
+  }
+
+  @Test
+  public void defaults() {
+    StressTestClient client = new StressTestClient();
+    assertEquals(singletonList(new InetSocketAddress("localhost", 8080)), client.addresses());
+    assertTrue(client.testCaseWeightPairs().isEmpty());
+    assertEquals(-1, client.durationSecs());
+    assertEquals(1, client.channelsPerServer());
+    assertEquals(1, client.stubsPerChannel());
+    assertEquals(8081, client.metricsPort());
+  }
+
+  @Test
+  public void allCommandlineSwitchesAreSupported() {
+    StressTestClient client = new StressTestClient();
+    client.parseArgs(new String[] {
+        "--server_addresses=localhost:8080,localhost:8081,localhost:8082",
+        "--test_cases=empty_unary:20,large_unary:50,server_streaming:30",
+        "--test_duration_secs=20",
+        "--num_channels_per_server=10",
+        "--num_stubs_per_channel=5",
+        "--metrics_port=9090"
+    });
+
+    List<InetSocketAddress> addresses = Arrays.asList(new InetSocketAddress("localhost", 8080),
+        new InetSocketAddress("localhost", 8081), new InetSocketAddress("localhost", 8082));
+    assertEquals(addresses, client.addresses());
+
+    List<TestCaseWeightPair> testCases = Arrays.asList(
+        new TestCaseWeightPair(TestCases.EMPTY_UNARY, 20),
+        new TestCaseWeightPair(TestCases.LARGE_UNARY, 50),
+        new TestCaseWeightPair(TestCases.SERVER_STREAMING, 30));
+    assertEquals(testCases, client.testCaseWeightPairs());
+
+    assertEquals(20, client.durationSecs());
+    assertEquals(10, client.channelsPerServer());
+    assertEquals(5, client.stubsPerChannel());
+    assertEquals(9090, client.metricsPort());
+  }
+
+  @Test(timeout = 5000)
+  public void gaugesShouldBeExported() throws Exception {
+    int serverPort = TestUtils.pickUnusedPort();
+    int metricsPort = TestUtils.pickUnusedPort();
+
+    TestServiceServer server = new TestServiceServer();
+    server.parseArgs(new String[]{"--port=" + serverPort, "--use_tls=false"});
+    server.start();
+
+    StressTestClient client = new StressTestClient();
+    client.parseArgs(new String[] {"--test_cases=empty_unary:1",
+        "--server_addresses=localhost:" + serverPort, "--metrics_port=" + metricsPort,
+        "--num_stubs_per_channel=2"});
+    client.startMetricsService();
+    client.runStressTest();
+
+    // Connect to the metrics service
+    ManagedChannel ch = ManagedChannelBuilder.forAddress("localhost", metricsPort)
+        .usePlaintext(true)
+        .build();
+
+    MetricsServiceGrpc.MetricsServiceBlockingStub stub = MetricsServiceGrpc.newBlockingStub(ch);
```

Comment:
That seems to complicate things to me, as we have multiple RPCs in this test. Using the simple API of the blocking stub and specifying a global timeout on the test, seems much easier.